src/extension.ts

```diff
@@ -171,41 +171,121 @@ class RunLineInEditor {
+                // concatenate all lines into a single command
+                lineNumber ++;
+                command = command.replace("`", "") + this.StripComments(source.document.lineAt(lineNumber).text);
+            }
+            return command;
+        } 
+        else
+        {
+            // execute only the selected text
+            const selectionStart = source.selection.start;
+            const selectionEnd = source.selection.end;
+            if (selectionStart.line === selectionEnd.line)
+            {
+                return this.StripComments(source.document.getText(new Range(selectionStart, selectionEnd)));
+            }
+            else
+            {
+                command = this.StripComments(source.document.lineAt(selectionStart.line).text.substring(selectionStart.character));
+                for (let index = selectionStart.line+1; index <= selectionEnd.line; index++) {
+                    var line = this.StripComments(source.document.lineAt(index).text);
+                    if (line.startsWith("az"))
+                    {
+                        window.showErrorMessage<any>("Multiple command selection not supported");
+                        return "";
+                    }
+                    if (index === selectionEnd.line)
+                    {
+                        command = command.replace("`", "") + line.substring(0, selectionEnd.character);
+                    }
+                    else
+                    {
+                        command = command.replace("`", "") + line;
+                    }
                 }
-            }, 50);
+                return command;
+            }
         }
-        this.commandRunningStatusBarItem.show();
-        clearTimeout(this.hideStatusBarItemTimeout);
-
-        this.parsedResult = undefined;
-        this.query = undefined; // TODO
-        const cursor = source.selection.active;
-        const line = source.document.lineAt(cursor).text;
-        return this.findResultDocument()
-            .then(document => window.showTextDocument(document, ViewColumn.Two, true))
-            .then(target => replaceContent(target, JSON.stringify({ 'Running command': line }) + '\n')
-                .then(() => exec(line))
-                .then(({ stdout }) => stdout, ({ stdout, stderr }) => JSON.stringify({ stderr, stdout }, null, '    '))
-                .then(content => replaceContent(target, content)
-                        .then(() => this.parsedResult = JSON.parse(content))
-                        .then(undefined, err => {})
-                )
-                .then(() => this.commandFinished(t0))
-            )
-            .then(undefined, console.error);
+    }
+
+    private StripComments(text: string)
+    {
+        // allow for single line comments on the same line as the command (// or #)
+        var i = text.search("//");
+        if (i !== -1)
+        {
+            return text.substring(0, i)
+        }
+        i = text.search("#");
```

Comment:
Can # be part of the command? Like in a string?

Reply:
Good point; it is possible to have strings in a JMESPath query. What do you think of using starts with "#" to indicate a comment?

Reply:
That's hard to get correct for all cases. One idea though: Have you tried collecting the multi-line command including newlines and comments and execute it like that? I think the way we use `cp.exec` spawns a shell and passes the command to the shell, so the shell might do the parsing of multi-lines and comments for you.

Reply:
I'll have to think on this some. The az cli gets very particular about the format of the command; it doesn't like newlines & doesn't like unexpected text (like comments).

Reply:
I think because it passes through the shell before getting to `az`, the shell will take care of removing comments and parsing the parameters into a list. Haven't tried this though.

Reply:
Ok, I think I've got this sorted out. It's not very elegant but I can live with it.